### DISCUSSION TITLE
[Q&A] How to restore chats history when domain change? | 当域名发生变化时，如何找回历史记录？

### DISCUSSION THREAD

**Discussion by JoshAtticus** [lobehub/lobehub#1127] [General Question] [answered]
### 💻 Operating System

macOS

### 📦 Environment

Other

### 🌐 Browser

Chrome

### 🐛 Bug Description

Recently, chat.aivvm.com started redirecting to lobe.caifree.com, so now I can't get my chats back! Please help!

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
you need to contact the author of `chat.aivvm.com`. Because the data is saved on `chat.aivvm.com` host's indexedDB.

You need to open chat page on `chat.aivvm.com` and then export the data. And then import them on `lobe.caifree.com`.

**Comment by JoshAtticus** [comment]
> you need to contact the author of `chat.aivvm.com`. Because the data is saved on `chat.aivvm.com` host's indexedDB.
> 
> You need to open chat page on `chat.aivvm.com` and then export the data. And then import them on `lobe.caifree.com`.

I can't open the chat page since it automatically redirects to the new site

**Comment by JoshAtticus** [comment]
Edit: I moved the indexeddb files from chat.aivvm.com to lobe.caifree.com, it worked!

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
> Edit: I moved the indexeddb files from chat.aivvm.com to lobe.caifree.com, it worked!

Amazing! How do you make it? Maybe this canbe  a guide for similar users.

**Comment by JoshAtticus** [comment, marked as answer]
> > Edit: I moved the indexeddb files from chat.aivvm.com to lobe.caifree.com, it worked!
> 
> Amazing! How do you make it? Maybe this canbe a guide for similar users.

**EDIT: important update!!!**

On Firefox,
1. Open your profile folder (C:\Users\<username>\AppData\Roaming\Mozilla\Firefox\Profiles\ on Windows and ~/Library/Application Support/Firefox/Profiles/ on macOS)
2. Go through the folders until you find the one with the storage folder 
3. Locate the folder for the old page and open it 
4. Open the idb folder inside of it and copy everything in there 
5. Find the folder for the new page and open it 
6. Go into the idb folder, make a backup of everything in there to somewhere safe
7. Paste the files from the old site and click yes when asked to replace all files and folders.
8. Once your chats appear, **do not refresh the page**, export all your data as soon as possible. If they don't appear, click the default chat.
9. Delete everything in the idb folder for the new page
10. Refresh the page and import your data

**This will erase all the stuff from the new page, which is why you make a backup before replacing the files**
